### ISSUE TITLE
No exoplayer version (View only)

### ISSUE THREAD

**Issue by omkar-tenkale** [vkay94/DoubleTapPlayerView#8]
How can i use this library as simple as adding it in constraint layout and set double tap listeners
while it animates the beautiful ripples without caring if its a video being played or anything else

**Comment by vkay94** [comment]
This library is designed for video playback, so I won't add this kind of extra View to it. Although it is not hard to implement such effect to other 'trigger' views. You have to override `onTouchEvent` / call `setOnTouchListener` on the View which should react to the double tapping (in this library it's  ExoPlayer's `PlayerView`) yourself though.

If you'd like to I can create a side-branch with this View since it's basically the same as the YouTube effect without the Player and SecondsView. Just give me some more info :)

**Comment by omkar-tenkale** [comment]
Sure
That'd be great


This view could be used as 
1.page navigator for PDF viewer
2.android videoview overlay
3.Music player seek helper

The custom view  **DoubleTapRippleView** can be used as

<FrameLayout>
<com.github.vkay94.dtpv.DoubleTapRippleView id="pdfPageSeeker"/>
<PDFView/>
</FrameLayout>


```
DoubleTapRippleView animatedTapDetector = findviewbyid(...)

animatedTapDetector.setTapListener(
void onClick(int gravity(left/right)){
   pdfView.toggleOptionsOverlayVisibility();
}
void on DoubleTap(int gravity (left/right)){
   if (gravity == left){
      pdfView.navigatePreviousPage()
   }else{
   pdfView.navigateNextPage()
  }
}
)
```
```

**Comment by vkay94** [comment]
Hey @omkar-tenkale , as promised I've written the View class with your requested functionality/callbacks (for this moment locally). 
You can check the demo here: [dt_ripple_demo_test_001.zip](https://github.com/vkay94/DoubleTapPlayerView/files/5146976/dt_ripple_demo_test_001.zip).

The usage is as follows: 

**Layout**
```
<ConstraintLayout android:id="@+id/root">
    <!-- Your other views -->
    <DoubleTapRippleView android:id="@+id/overlay" />
</ConstraintLayout>
```

**Code (Kotlin)** (excerpt from the demo)

```
// root is the container which reacts to the gestures
root.setOnTouchListener(overlay)

overlay.stayDelay(700)
    .performListener(object : DoubleTapRippleView.PerformListener {
        override fun onAnimationStart() {
            overlay.visibility = View.VISIBLE
        }

        override fun onAnimationEnd() {
            overlay.visibility = View.INVISIBLE
        }

    })
    .tapListener(object : DoubleTapRippleView.TapListener {
        override fun onSingleTap(gravity: Int) {
            val text = when (gravity) {
                DoubleTapRippleView.LEFT -> "onSingleTap left"
                DoubleTapRippleView.RIGHT -> "onSingleTap right"
                else -> "Clicked else"
            }

            lastActionTextView.text = "Last action: $text"
        }

        override fun onDoubleTap(gravity: Int) {
            when (gravity) {
                DoubleTapRippleView.LEFT -> counter -= 1
                DoubleTapRippleView.RIGHT -> counter += 1
            }

            val text = when (gravity) {
                DoubleTapRippleView.LEFT -> "onDoubleTap left"
                DoubleTapRippleView.RIGHT -> "onDoubleTap right"
                else -> "Clicked else"
            }

            lastActionTextView.text = "Last action: $text"
            textView.text = counter.toString()
        }
    })
```

**Added methods to the Ripple View:**
- `fun multiDoubleTap(enabled: Boolean)` : if true then each ongoing tap is handled as double tapped (the same as `YoutubeOverlay`), if false only*real* double taps are handled (4 fast taps = 2 double taps; 3 fast taps = 1 double tap + 1 single tap)
- `fun stayDelay(duration: Long)`: Basically the same as `doubleTapDelay` of `DoubleTapPlayerView`

Let me know if it's as expected or you want to have changed, otherwise I'd push it to the branch ;)

**Comment by omkar-tenkale** [comment]
Seems flawless. Just as imagined. 
Eagerly waiting to add it to my projects. 
Hope it'll benefit many people. 

Thanks for the efforts 🤗

**Comment by vkay94** [comment]
Hey, sorry for this delay, it took some time to *optimize* some logic (I had to change the structure because of some touch events, so that it doesn't block other gestures ... mostly).

onInterceptTouchEvent is not cool to handle in my opinion o.0

I've also pushed the code with the example and updated Readme (read it carefully). I had to remove onSingleTap, so that onSingleTap is equal to onClick of the underlying Views.

Enough theory, here are the instructions: [Link](https://github.com/vkay94/DoubleTapPlayerView/tree/ripple_view). I hope it matches your requirements, otherwise let me know :D
